### PR TITLE
Dlg pca

### PR DESCRIPTION
**Pull request by lilyclements**

### CODE REVIEW

**Comment by lilyclements** [comment]
@africanmathsinitiative/instat This is now ready for review

File: instat/dlgPrincipalComponentAnalysis.vb [hunk 1]
```diff
@@ -13,150 +13,134 @@
 '
 ' You should have received a copy of the GNU General Public License k
 ' along with this program.  If not, see <http://www.gnu.org/licenses/>.
+Imports instat
 Imports instat.Translations
 Public Class dlgPrincipalComponentAnalysis
     Public bFirstLoad As Boolean = True
+    Private bReset As Boolean = True
+    Private bResetSubdialog As Boolean = False
     Public ExplanatoryVariables
```
Comment (stevekogo) [inline instat/dlgPrincipalComponentAnalysis.vb]: This is partially defined and not used anywhere. You need to take it out.

Comment (lilyclements) [inline instat/dlgPrincipalComponentAnalysis.vb]: Thanks, I've commented it out for the time being in case it helps later on with the sub

**Comment by lilyclements** [comment]
@stevekogo Is this okay now?

File: instat/dlgPrincipalComponentAnalysis.vb (hunk 1 above)
Comment (stevekogo) [inline instat/dlgPrincipalComponentAnalysis.vb]: I tried testing but the subdialog needs strmodelname defined. You may need to uncomment it.

Comment (lilyclements) [inline instat/dlgPrincipalComponentAnalysis.vb]: Accidentally commented them both out which is why there's two commits. Only Explanatory Variables is commented out, I've uncommented the strModelName. 

How is this looking now?

**Comment by dannyparsons** [comment]
@stevekogo I think this is ready for a re-review.

**Comment by lilyclements** [comment]
@stevekogo Would you mind re-reviewing this? Thanks

**Comment by stevekogo** [comment]
@lilyclements I thought I reviewed thison tuesday but am surprised not seeing my comments, maybe internet connection was the problem.

I tried conducting PCA and it seems the dialog was fine during the default running. Upon clicking the command link to subdialog an error occurs where the three display results(eigen values, eigen vectors and rotaion) fails to get the model name. Here is the error that I received.
capture

**Comment by lilyclements** [comment]
This is because the sub dialog is not yet set up. I've removed the link between the sdg and dlg now so @stevekogo this is ready for a review

**Comment by lilyclements** [comment]
@stevekogo This is waiting for a re-review still :)

**Comment by stevekogo** [comment]
The dialog is well implemented but I think getting it working with the subdialog will be a good progress.

**Comment by stevekogo** [comment]
I think you will have to get the dialog work with the subdialog so that the PCA procedures can be performed once this is merged, else not much can be done with the dialog defaults.

**Comment by lilyclements** [comment]
That's a good point, it's more functional as it is in the copy and we're having a new version soonj so should keep this out of it if this is not complete.
I'm a little stuck on the sub dialog, however, I've noticed you are looking at Correlations and doing the dialog there so I plan to get started on this sub when I'm able to look over the correlations sub

**Comment by Lunalo** [comment]
@lilyclements 

Have you looked at this?

**Comment by lilyclements** [comment]
Not yet, I'm unsure at the moment of the subdialogs in the new system so waiting until I can get that sorted until this is ready for review/merge

**Comment by lilyclements** [comment]
This dialog is currently dormant as I am unsure how to implement its sub in the new system

**Comment by stevekogo** [comment]
@lilyclements Could you describe the problem clearly so that people can understand it and probly help or give suggestions.

**Comment by lilyclements** [comment]
Yes, so I have not yet got around to implementing the sub dialog of this dialog, and so the dialog has not been merged yet. I will work on the sub dialog soon.
If anyone plans to start implementing the sub, then feel free to make a copy of this branch and work on it.

**Comment by maxwellfundi** [comment]
@africanmathsinitiative/developers is anyone taking this or @lilyclements its still in your plans?

**Comment by lilyclements** [comment]
@maxwellfundi I haven't got around to the sub dialog for this yet. If someone wished to make a copy of this branch and work on the sub dialog then they can

**Comment by dannyparsons** [comment]
Can this be closed? There's conflicts so I assume it's been updated somewhere else?

**Comment by lilyclements** [comment]
No, the conflicts are because we updated the package name. This still hasn't been fully updated anywhere

**Pull request closed without merging**